multus/multus_test.go

```diff
@@ -68,6 +74,56 @@ type fakeExec struct {
 	plugins  []*fakePlugin
 }
 
+type FakeKubeClient struct {
+    Client           kubernetes.Interface
+	NetClient        netclient.K8sCniCncfIoV1Interface
+	EventBroadcaster record.EventBroadcaster
+	EventRecorder    record.EventRecorder
+	ErrorCode 		 error // may need to be looked at...
+}
+
+// FakeKubeClient implements ClientInfo
+// var _ ClientInfo = &FakeKubeClient{}
+
+// AddPod adds pod into kubernetes
+func (c *FakeKubeClient) AddPod(pod *v1.Pod) (*v1.Pod, error) {
+	return c.Client.Core().Pods(pod.ObjectMeta.Namespace).Create(pod)
+}
+
+// GetPod gets pod from kubernetes
+func (c *FakeKubeClient) GetPod(namespace, name string) (*v1.Pod, error) {
+	pod, _ := c.Client.Core().Pods(namespace).Get(name, metav1.GetOptions{})
+	return pod, errors.NewServiceUnavailable("Testing purposes")
```

Comment:
This code always return error. How about to have a function which sets error code to be returned, such like that:
`func (c *FakeKubeClient) SetError(err *StatusError)`

Of corse, if we set `SetError(nil)` then this client returns no error (i.e. succeed to get pod).